src/test/java/com/spotify/dns/SimpleLookupFactoryTest.java

```diff
@@ -46,18 +47,13 @@ public void shouldCreateLookups() {
   }
 
   @Test
-  public void shouldCreateNewLookupsEachTime() {
+  public void shouldNotCreateNewLookupsEachTime() {
```

Comment:
Split this test into forName / sessionForName calls. The method name doesn't make sense if combined